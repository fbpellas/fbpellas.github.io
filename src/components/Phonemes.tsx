import * as React from 'react';
import { BASE_PATH_IMG, IPA_DEFINITION } from '../constants';

export interface PhonemesProps {
  setPageAndClear: (hash: string) => void
}

const Phonemes: React.FC<PhonemesProps> = (props) => {
  const { setPageAndClear } = props;

  return (
    <div className="block-2">
      <div className="article">
        <h3 className="h3-title">Phonemes</h3>
        <div>
          According to the {IPA_DEFINITION} (IPA), there are 26 letters and 44 phonemes (or sounds)
          in the English alphabet. These letters are divided into two categories:{' '}
          <a className="clickable-page" href="#vowels" onClick={() => setPageAndClear('vowels')}>
            vowels
          </a>{' '}
          and{' '}
          <a className="clickable-page" href="#consonants" onClick={() => setPageAndClear('consonants')}>
            consonants
          </a>
          .
        </div>
        <br />
        <div>
          <div className="note">Note:</div> Phonemes should not rely on the word’s spelling. For example, the word m
          <i>
            <b>oo</b>
          </i>
          n is not spelled with the letter ‘u’, yet is produced with the long /u/ phoneme.{' '}
        </div>
        <img
          className="margin-top half-img"
          src={`${BASE_PATH_IMG}IPA chart.jpg`}
          alt="Learn the Art of Speaking American English"
        />
      </div>
    </div>
  );
}

export { Phonemes }
