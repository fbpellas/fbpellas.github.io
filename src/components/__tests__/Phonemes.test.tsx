import React from 'react';
import { render } from '@testing-library/react';
import { Phonemes } from '../Phonemes';

test('Phonemes', () => {
  const fn = jest.fn();
  const { container } = render(<Phonemes setPageAndClear={fn} />);
  expect(container).toMatchSnapshot();
});
