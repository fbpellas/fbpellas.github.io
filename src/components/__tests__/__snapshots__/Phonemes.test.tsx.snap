// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Phonemes 1`] = `
<div>
  <div
    class="block-2"
  >
    <div
      class="article"
    >
      <h3
        class="h3-title"
      >
        Phonemes
      </h3>
      <div>
        According to the 
        International Phonetic Alphabet
         (IPA), there are 26 letters and 44 phonemes (or sounds) in the English alphabet. These letters are divided into two categories:
         
        <a
          class="clickable-page"
          href="#vowels"
        >
          vowels
        </a>
         
        and
         
        <a
          class="clickable-page"
          href="#consonants"
        >
          consonants
        </a>
        .
      </div>
      <br />
      <div>
        <div
          class="note"
        >
          Note:
        </div>
         Phonemes should not rely on the word’s spelling. For example, the word m
        <i>
          <b>
            oo
          </b>
        </i>
        n is not spelled with the letter ‘u’, yet is produced with the long /u/ phoneme.
         
      </div>
      <img
        alt="Learn the Art of Speaking American English"
        class="margin-top half-img"
        src="https://raw.githubusercontent.com/fbpellas/fbpellas.github.io/main/public/img/IPA chart.jpg"
      />
    </div>
  </div>
</div>
`;
